src/parser/jobs/rdm/modules/MeleeCombos.js

```diff
@@ -0,0 +1,562 @@
+				let newHigherMana = higherMana
+
+				// Determine how much mana would be wasted to cap with this delay, then adjust post-delay mana totals to cap before further comparisons
+				const manaLoss = Math.max(newLowerMana - MANA_CAP, 0) + Math.max(newHigherMana - MANA_CAP, 0)
+				newLowerMana = Math.min(newLowerMana, MANA_CAP)
+				newHigherMana = Math.min(newHigherMana, MANA_CAP)
+
+				if (newHigherMana < newLowerMana) {
+					// Mana rebalancing resulted in the original higherMana becoming the lower total (guaranteed proc), valid option, push onto stack
+					possibleDelays.push({
+						finisher: [lowerManaActions.proc, lowerManaActions.dualcast, higherManaActions.finisher],
+						manaLoss: manaLoss,
+					})
+				} else {
+					// Verify that using the finisher of higherMana won't put us out of balance at the end
+					const finisherManaGain = MANA_GAIN[higherManaActions.finisher.id].white || MANA_GAIN[higherManaActions.finisher.id].black
+					if (!((newHigherMana + finisherManaGain - newLowerMana) > MANA_DIFFERENCE_THRESHOLD)) {
+						// This is a net gain whether or not acceleration would be available - we can now fish for an additional proc of higherMana, push onto stack
+						possibleDelays.push({
+							finisher: [lowerManaActions.proc, lowerManaActions.dualcast, ACTIONS.ACCELERATION, higherManaActions.finisher],
+							manaLoss: manaLoss,
+						})
+					}
+				}
+			}
+		} else {
+			// These cases should only be hit if lowerMana == higherMana (we were in balance at start of combo), to test benefits of delaying combo to imbalance mana
+			// If lowerManaProc isn't available and lowerMana < higherMana, recommendation will always be the lowerManaActions.finisher
+			if (higherManaProc) { // eslint-disable-line no-lonely-if
+				let newLowerMana = lowerMana + (MANA_GAIN[lowerManaActions.dualcast.id].white || MANA_GAIN[lowerManaActions.dualcast.id].black)
+				let newHigherMana = higherMana + (MANA_GAIN[higherManaActions.proc.id].white || MANA_GAIN[higherManaActions.proc.id].black)
+
+				// Determine how much mana would be wasted to cap with this delay, then adjust post-delay mana totals to cap before further comparisons
+				const manaLoss = Math.max(newLowerMana - MANA_CAP, 0) + Math.max(newHigherMana - MANA_CAP, 0)
+				newLowerMana = Math.min(newLowerMana, MANA_CAP)
+				newHigherMana = Math.min(newHigherMana, MANA_CAP)
+
+				if (newHigherMana < newLowerMana) {
+					// Mana rebalancing resulted in the original higherMana becoming the lower total (guaranteed proc), valid option, push onto stack
+					possibleDelays.push({
+						finisher: [higherManaActions.proc, lowerManaActions.dualcast, higherManaActions.finisher],
+						manaLoss: manaLoss,
+					})
+				}
+				// Disregard case when newHigherMana remains even or higher than newLowerMana - error in calling the delay module
+				console.log(`Delay module called with ${lowerMana}|${higherMana} starting mana, no lowerManaProc, and a higherManaProc.`)
```

Comment:
missed some logging~
if it's a theoretically impossible case, it should be triggering `brokenLog` - if it's not, it can either be squashed or presented to the user in some manner.

Reply:
I think it was mostly my sheet anchor to windward, especially before I had the melee combo logic test cases to compare against.  Removed.